### ISSUE TITLE
Corrupted/lost text when scrolling in vertically split neovim window using wezterm with `wezterm` TERM enviornment variable

### ISSUE THREAD

**Issue by CarbonChauvinist** [zellij-org/zellij#3412]
# 1. Graphical issue inside a terminal pane (eg. something does not look as it should)

Please attach the files that were created in `/tmp/zellij-1000/zellij-log/` to the extent you are comfortable with.

[zellij-12.log](https://github.com/user-attachments/files/15755270/zellij-12.log)
[zellij.log](https://github.com/user-attachments/files/15755271/zellij.log)

**Basic information**

`zellij --version`: `zellij 0.41.0`

`stty size`: `32 145`

`uname -av`: `Linux lap 6.9.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 31 May 2024 15:14:45 +0000 x86_64 GNU/Linux`

## Further information

After further testing using the terminals I use this only occurs with Wezterm when run with wezterm TERM enviornment variable. This may be a Wezterm issue?

Wezterm version: `wezterm 20240203-110809-5046fc22`

The terminals I tested where the issue did not present itself: (terminal/TERM)
* Foot with foot-extra TERM env var
* Sakura with xterm-256color TERM env var
* Wezterm with xterm-256color TERM env var

Verified this issue is present on the latest build of zellij (using zellij-git package from aur). Currently using the development version of neovim (NVIM v0.11.0-dev-199+g4881211097).

When opening neovim and creating a vertical split on any file within neovim, scroll down in one window within neovim and observe that the other split window begins to scroll as well and then eventually becomes garbled and loses text. One can switch back and forth between windows and scroll up and down and observe the same behavior on the opposite window from the one currently being scrolled.

To reproduce:

0. run wezterm with term set to `wezterm` in [config](https://wezfurlong.org/wezterm/config/lua/config/term.html?h=term)
1. run zellij with default settings.
2. run development version of neovim also in a clean state with no plugins loaded (i.e. `nvim --clean -u NONE -i NONE`)
3. Open any file
4. create a vertical split
5. Scroll down in the current window and observe the reported issue in the neighboring window within neovim

Attached a short screen capture of the behavior for reference. 

Also note,  in Wezterm with `wezterm` TERM env this issue does not present itself when running latest version of vim available in the standard Arch repos, nor does it present itself when running neovim without zellij.

May be related to issue #263

https://github.com/zellij-org/zellij/assets/21365869/c1ab6a34-616f-4365-ad2a-6cc5d45ca6b7

**Comment by mike-lloyd03** [comment]
I'm having this problem as well and it only happens inside Zellij and nvim. However, when downgrading to 40.0, it's still happening.

I think this relates to nvim 0.10.0. Downgrading to nvim 0.9.5 fixes the problem for me.

**Comment by mike-lloyd03** [comment]
I take it back. This appears to be happening in nvim 0.9.5 and Zellij 0.40.0. I'm noticing it in Wezterm and Kitty.

**Comment by mike-lloyd03** [comment]
Okay I did some more testing. It looks like this is limited to Zellij 0.40.1. If I downgrade to 0.40.0 and `delete-all-sessions`, the problem disappears.

**Comment by mrdgo** [comment]
Have you guys tried scrolling with Ctrl-u and Ctrl-d? Probably boils down to the same bug but it looks even worse. Please confirm, otherwise I'll open a separate issue.

**Comment by CarbonChauvinist** [comment]
@mrdgo Yes <C-u> and <C-d> present the same behavior. The behavior of the corrupted pane is never really consistent, but I can always reproduce the issue. Again, it only occurs with wezterm and when I've set the wezterm terminfo.

I've attached another screen grab which shows this in action - apologies for some of the artifacts/compression that's not related to the issue at hand but rather the screen recording itself. In this I scroll down first line by line (i.e. `j` and `k`) and then by page (i..e. `<C-d>` and `<C-u>`) in the unaffected nvim window. I then switch the affected nvim window and do the same, scroll by line and then by page. I also jump between beginning and end of buffer with `gg` and `G`.


https://github.com/zellij-org/zellij/assets/21365869/2668cba7-60f5-476c-a81d-b91f12e65aae

**Comment by schrieveslaach** [comment]
It seems that using `:InspetTree` in Neovim is very related and maybe, a good candidate to iterate on the issue:

https://github.com/user-attachments/assets/08ab75b0-2f1e-4eb6-ba18-fdadc06479bc

**Comment by schrieveslaach** [comment]
A workaround for me is to set `TERM=xterm`  whenever I start nvim (see https://gitlab.com/schrieveslaach/dotfiles/-/commit/29ae9ff6c5cb5c82c995cfe6b4c6d61890e76a24)

**Comment by Spuxy** [comment]
upstream of wezterm says @schrieveslaach, that we should use xterm https://wezterm.org/config/lua/config/term.html
